src/main/java/forestry/plugins/compat/PluginNatura.java

```diff
@@ -72,45 +73,92 @@ public String getFailMessage() {
+			if(itemName.matches("^edibles|.*fruit_item|soups$")) {
+				complexConsumeSubItems(item, "edibles", subitem -> {
+					final String subItemName = subitem.getUnlocalizedName();
+					if(subItemName.matches("^.*berry$")) {
+						berries.add(subitem);
+						return;
+					}
+
+					if(subItemName.matches("^.*(fruit_item|apple)$")) {
+						fruits.add(subitem);
+						return;
+					}
+
+					if(subItemName.matches("^.*stew$")) {
+						soups.add(subitem);
+						return;
+					}
+
+					edibles.add(subitem);
+				});
+			}
 
-		ArrayList<String> saplingItemKeys = new ArrayList<>();
+			if(itemName.matches("^.*_seeds$")) {
+				consumeSubItems(item, "seeds", seeds);
+				return;
+			}
 
-		if (saplingNatura != null) {
-			saplingItemKeys.add("florasapling");
-		}
-		if (saplingNaturaRare != null) {
-			saplingItemKeys.add("Rare Sapling");
-		}
+			if(itemName.matches("^.*_logs\\d?$")) {
+				consumeSubItems(item, "logs", logs);
+				return;
+			}
 
-		for (String key : saplingItemKeys) {
-			Item saplingItem = GameRegistry.findItem(NATURA, key);
+			if(itemName.matches("^.*_crop\\d?$")) {
+				consumeSubItems(item, "cropBlocks", cropBlocks);
+				return;
+			}
 
-			ItemStack saplingWild = new ItemStack(saplingItem, 1, OreDictionary.WILDCARD_VALUE);
-			RecipeUtil.addFermenterRecipes(saplingWild, ForestryAPI.activeMode.getIntegerSetting("fermenter.yield.sapling"), Fluids.BIOMASS);
+			if(itemName.matches("^materials$")) {
+				complexConsumeSubItems(item, "materials", subitem -> {
+					final String subItemName = subitem.getUnlocalizedName();
 
-			String saplingName = ItemStackUtil.getItemNameFromRegistryAsString(saplingItem);
-			FMLInterModComms.sendMessage(Constants.MOD_ID, "add-farmable-sapling", String.format("farmArboreal@%s.-1", saplingName));
-		}
+					if (subItemName.matches("^.*(barley|cotton)$")) {
+						crops.add(subitem);
+						return;
+					}
+					materials.add(subitem);
+				});
+			}
+		});
+	}
 
-		Item netherBerry = GameRegistry.findItem(NATURA, "berry.nether");
-		Item berry = GameRegistry.findItem(NATURA, "berry");
-		Item barley = GameRegistry.findItem(NATURA, "barleyFood");
-		
-		berryBlight = new ItemStack(netherBerry, 1, 0);
-		berryDusk = new ItemStack(netherBerry, 1, 1);
-		berrySky = new ItemStack(netherBerry, 1, 2);
-		berrySting = new ItemStack(netherBerry, 1, 3);
-		berryRasp = new ItemStack(berry, 1, 0);
-		berryBlue = new ItemStack(berry, 1, 1);
-		berryBlack = new ItemStack(berry, 1, 2);
-		berryMalo = new ItemStack(berry, 1, 3);
-		itemBarley = new ItemStack(barley);
+	private void consumeSubItems(Item item, String groupName, Collection<ItemStack> consumer) {
+		complexConsumeSubItems(item, groupName, subItem -> consumer.add(subItem));
+	}
+
+	private void complexConsumeSubItems(Item item, String groupName, Consumer<ItemStack> consumer) {
+		final ArrayList<ItemStack> subItems = new ArrayList<>();
+		item.getSubItems(item, CreativeTabs.SEARCH, subItems);
```

Comment:
Well this breaks everything. Since iterating over 2^16-1 possible subItems is just too much to be efficient. And as far as I see there is no other simple way to get sub items on the server side.
I would postpone this change to 1.12.

Since a lot of the code depends on that function, this is killing Natura support for 1.10.
Cause I will not add 60 hard coded items here.

Any suggestions?